js/src/hooks/useUpdateRestAPIAuthorizeStatusByUrlQuery.js

```diff
@@ -32,6 +37,16 @@ const useUpdateRestAPIAuthorizeStatusByUrlQuery = () => {
 
 	const handleUpdateRestAPIAuthorize = useCallback( async () => {
 		try {
+			if ( ! googleWPCOMAuthNonce || ! nonceFromURLQuery ) {
+				return;
+			}
+
+			if ( nonceFromURLQuery !== googleWPCOMAuthNonce ) {
+				throw new Error(
+					'Nonces mismatch when updating REST API authorize status.'
+				);
+			}
```

Comment:
I wouldn't do the nonce verification in the Frontend as can be easily altered in the client side. What about doing the check in the backend inside  `PUT rest-api/authorize`? 

So we can send the query nonce and compare it in the backend with the saved in Options.